cmd/gateway-gcs.go

```diff
@@ -835,7 +835,11 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 
 	partZero.Delete(l.ctx)
 
-	return fromGCSObjectInfo(attrs), gcsToObjectError(traceError(err), bucket, key)
+	objInfo := fromGCSObjectInfo(attrs)
+	if objInfo.ETag == "" {
+		objInfo.ETag = bucket + key + uploadID
```

Comment:
you can do a md5Sum of the "bucket + key" - uploadID is gone after the upload is complete, so we don't need this to be part of the ETag. 

```
h := md5.New()
io.WriteString(objInfo.Name+objectInfo.Key+objectInfo.Size)
hex.EncodeToString(h.Sum(nil))
```

^^ this would be better.